stl/inc/random

```diff
@@ -2932,15 +2932,32 @@ private:
             _Ty _Vx1;
             _Ty _Vx2;
             _Ty _Sx;
-            for (;;) { // reject bad values
+            for (;;) { // reject bad values to avoid generating NaN/Inf on the next calculations
                 _Vx1 = 2 * _NRAND(_Eng, _Ty) - 1;
                 _Vx2 = 2 * _NRAND(_Eng, _Ty) - 1;
                 _Sx  = _Vx1 * _Vx1 + _Vx2 * _Vx2;
-                if (_Sx < 1) {
+                if (_Sx < _Ty{1} && _Vx1 != _Ty{0} && _Vx2 != _Ty{0}) {
+                    // good values!
                     break;
                 }
             }
-            const auto _Fx = static_cast<_Ty>(_CSTD sqrt(-2.0 * _CSTD log(_Sx) / _Sx));
+
+            _Ty _LogSx;
+            if (_Sx > 1.0e-4) {
+                _LogSx = _STD log(_Sx);
+            } else {
+                // Bad _Sx value! Very small values will overflow log(_Sx)/_Sx
+                // generating a new value based on scaling method.

```

Comment:
Would this be clearer? We have comments of the form `Full sentence. Another full sentence.` (with capitalization and punctuation) and comments of the form `sentence fragment` (no capitalization, no period), which are both fine. However, it's confusing when sentences and sentence fragments are mixed without any punctuation.
```suggestion
                // Bad _Sx value! Very small values will overflow log(_Sx) / _Sx.
                // Generate a new value based on scaling method.
```

Reply:
Sorry I was a little late. Thanks for your feedback and also applying it. It's always very difficult for me to follow English grammar and the repo rules on writing comments. Unfortunately, English is not my native language. I'm going to try my best on the next PRs. :)